src/app/components/create-dataset-modal/create-dataset-modal.component.ts

```diff
@@ -0,0 +1,367 @@
+  }],
+  ['BASIC', {
+    organization: 'PS',
+  }],
+  ['LARGE', {
+    organization: 'PS',
+  }]
+]);
+
+@Component({
+  selector: 'create-dataset-modal',
+  templateUrl: './create-dataset-modal.component.html',
+  styleUrls: ['./create-dataset-modal.component.scss',
+  '../../../../src/app/shared/modal.component.scss']
+})
+export class CreateDatasetModal {
+  private properties = {
+    template: '',
+    name: '',
+    allocationUnit: '',
+    averageRecordUnit: '',
+    primarySpace: '',
+    secondarySpace: '',
+    directoryBlocks: '',
+    recordFormat: '',
+    recordLength: '',
+    blockSize: '',
+    datasetNameType: '',
+    organization: ''
+  };
+  private numericPattern: string;
+  private numericPatternExZero: string;
+  private datasetNamePattern: string;
+  private alphaNumericPattern: string;
+  private templateOptions: string[];
+  private allocationUnitOptions: string[];
+  private datasetNameTypeOptions: string[];
+  private recordFormatOptions: string[];
+  private organizationOptions: string[];
+  private recordUnitOptions: string[];
+  private matcher = new CustomErrorStateMatcher();
+  private isDirBlockValid: boolean = true;
+  private dirBlockError: string;
+  private dirBlockTouched: boolean = false;
+  private isPrimeSpaceValid: boolean = true;
+  private isSecondSpaceValid: boolean = true;
+  private isRecLengthValid: boolean = true;
+  private isBlockSizeValid: boolean = true;
+  private primarySpaceError: string = "Primary space value cannot be more than '16777215' ";
+  private secondarySpaceError: string = "Secondary space value cannot be more than '16777215' ";
+  private recordLengthError: string = "Record length cannot be more than '32760' bytes";
+  private blockSizeError: string = "Block size cannot be more than '32760' bytes";
+  private isRecordFormatValid: boolean = true;
+  private recordFormatErrorMessage: string;
+  private blockSizeTouched: boolean = false;
+  private isAllocUnitValid: boolean = true;
+  private allocUnitErrorMessage: string = "If allocation unit is 'BLKS' then record format must be blocked type: FB, VB, VBA";
+  private isRecFormatTouched: boolean = false;
+
+  @ViewChild('dirblocks') dirblocks: ElementRef;
+  @ViewChild('primeSpace') primeSpace: ElementRef;
+  @ViewChild('allocUnit') allocUnit: MatSelect;
+  @ViewChild('secondSpace') secondSpace: ElementRef;
+  @ViewChild('recordLength') recordLength: ElementRef;
+  @ViewChild('recordFormat') recordFormat: MatSelect;
+  @ViewChild('dsorg') dsorg: ElementRef;
+
+
+  constructor(private el: ElementRef,) { }
+
+  ngOnInit() {
+    this.numericPattern = "^[0-9]*$";
+    this.numericPatternExZero = "^[1-9][0-9]*$";
+    this.datasetNamePattern = "^[a-zA-Z#$@][a-zA-Z0-9#$@-]{0,7}([.][a-zA-Z#$@][a-zA-Z0-9#$@-]{0,7}){0,21}$";
+    this.alphaNumericPattern = "^[a-zA-Z0-9]*$";
+    this.templateOptions = ['JCL','COBOL','PLX', 'XML'];
+    this.allocationUnitOptions = ['BLKS','TRKS','CYLS', 'KB', 'MB', 'BYTES', 'RECORDS'];
```

Comment:
the API currently uses TRK, CYL
since that code is already merged, i dont want to change it because it would be technically breaking change.
can you just change these to remove the "S"?